app/src/androidTest/java/com/swent/suddenbump/ui/overview/OverviewScreenTest.kt

```diff
@@ -51,18 +100,31 @@ class OverviewScreenTest {
     composeTestRule.onNodeWithTag("settingsFab").assertIsDisplayed()
     composeTestRule.onNodeWithTag("seeFriendsFab").assertIsDisplayed()
     composeTestRule.onNodeWithTag("appName").assertIsDisplayed()
-    composeTestRule.onNodeWithTag("distanceTitle").assertIsDisplayed()
+  }
+
+  @Test
+  fun displaysFriendsWithinCategories() {
+    composeTestRule.onRoot().printToLog("UI_TREE")
+    composeTestRule.onNodeWithTag("Within 5km").assertIsDisplayed()
```

Comment:
Let's improve the OverviewScreen tests by using conditional assertions for distance categories based on friend data availability. This will ensure that elements like "Within 5km" or "No friends nearby" are only asserted if applicable, making tests more robust and avoiding false failures. Something like the following code
`if (userFriendsFlow.value.any { userViewModel.getRelativeDistance(it) <= 5000f }) {
    composeTestRule.onNodeWithTag("Within 5km").assertIsDisplayed()
} else {
    composeTestRule.onNodeWithTag("Within 5km").assertDoesNotExist()
}

if (userFriendsFlow.value.isEmpty()) {
    composeTestRule.onNodeWithTag("noFriends").assertIsDisplayed()
}`